pytests/tests/test_list.py

```diff
@@ -0,0 +1,66 @@
+#
+# Copyright (C) 2019 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the GNU General Public License v2 (the "License");
+# you may not use this file except in compliance with the License. The terms
+# of the License are located in the COPYING file of this distribution.
+#
+#   Author: Siddharth Chandrasekaran <csiddharth@vmware.com>
+
+import pytest
+
+@pytest.fixture(scope='module', autouse=True)
+def setup_test(utils):
+    yield
+    teardown_test(utils)
+
+def teardown_test(utils):
+    pass
+
+def helper_test_list_sub_cmd(utils, sub_cmd):
+    ret = utils.run([ 'tdnf', 'list', sub_cmd ])
+    assert(ret['retval'] == 0)
+    ret = utils.run([ 'tdnf', 'list', sub_cmd, utils.config["sglversion_pkgname"] ])
+    assert(ret['retval'] == 0)
+    ret = utils.run([ 'tdnf', 'list', sub_cmd, 'invalid_package' ])
+    assert(ret['retval'] == 1011)
+
+def test_list_top(utils):
+    ret = utils.run([ 'tdnf', 'list' ])
+    assert(ret['retval'] == 0)
+    ret = utils.run([ 'tdnf', 'list', utils.config["sglversion_pkgname"] ])
+    assert(ret['retval'] == 0)
+    ret = utils.run([ 'tdnf', 'list', 'invalid_package' ])
+    assert(ret['retval'] == 1011)
+
+def test_list_sub_cmd(utils):
+    helper_test_list_sub_cmd(utils, 'all')
+    helper_test_list_sub_cmd(utils, 'installed')
+    helper_test_list_sub_cmd(utils, 'available')
+    helper_test_list_sub_cmd(utils, 'obsoletes')
+    helper_test_list_sub_cmd(utils, 'extras')
+    helper_test_list_sub_cmd(utils, 'recent')
+
+def test_list_upgrades(utils):
+    mpkg = utils.config["mulversion_pkgname"]
+    mpkg_version = utils.config["mulversion_lower"]
+    spkg = utils.config["sglversion_pkgname"]
+
+    ret = utils.run([ 'tdnf', 'list', 'upgrades' ])
+
+    # TODO Fix this; see issue #94
```

Comment:
Please make a note in the issue #94 to fix this corresponding tests as well.

Reply:
This comment you have here is reference enough (it shows up on that issue lising). Marking this as resolved.